cborg/src/Codec/CBOR/Decoding.hs

```diff
@@ -519,6 +564,13 @@ decodeBytes :: Decoder s ByteString
 decodeBytes = Decoder (\k -> return (ConsumeBytes (\bs -> k bs)))
 {-# INLINE decodeBytes #-}
 
+-- | Decode canonical representation of a string of bytes as a @'ByteString'@.
+--
+-- @since 0.2.1.0
+decodeBytesCanonical :: Decoder s ByteString
+decodeBytesCanonical = Decoder (\k -> return (ConsumeBytesCanonical (\bs -> k bs)))
```

Comment:
I don't think there's any danger of it being quadratic since this is not called in a loop. Honestly I'd have to check the core to see if it makes any difference, but I think in many cases the k will be known when adjacent decoders are connected with `>>=` and the `>>=` is inlined. So this should either disappear or make clear what the arity of k is. But as I say, to check properly one would need to go look at the core.